From 3dd7ba1084567c2f817861bcd151fd31a84eebe4 Mon Sep 17 00:00:00 2001
From: Andrey Semashev <andrey.semashev@gmail.com>
Date: Sat, 27 Jan 2024 01:21:38 +0300
Subject: [PATCH] Updated Boost.Filesystem usage.

Removed usage of Boost.Filesystem APIs that were deprecated and
then removed.
---
 include/boost/graph/distributed/adjlist/serialization.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/boost/graph/distributed/adjlist/serialization.hpp b/boost/graph/distributed/adjlist/serialization.hpp
index d6b14e5d..55fa4e8c 100644
--- a/boost/graph/distributed/adjlist/serialization.hpp
+++ b/boost/graph/distributed/adjlist/serialization.hpp
@@ -13,6 +13,7 @@
 # include <boost/lexical_cast.hpp>
 # include <boost/foreach.hpp>
 # include <boost/filesystem/path.hpp>
+# include <boost/filesystem/directory.hpp>
 # include <boost/filesystem/operations.hpp>
 # include <cctype>
 # include <fstream>
@@ -91,7 +92,7 @@ namespace detail { namespace parallel
 
           for (filesystem::directory_iterator i(filename), end; i != end; ++i)
           {
-              if (!filesystem::is_regular(*i))
+              if (!filesystem::is_regular_file(*i))
                   boost::throw_exception(std::runtime_error("directory contains non-regular entries"));
 
               std::string process_name = i->path().filename().string();
